Tests/UpstreamIntegrationTests/UpstreamIntegrationTests.swift

```diff
@@ -69,52 +66,495 @@ class UpstreamIntegrationTests: TestBase {
+
+        Edge.sendEvent(experienceEvent: experienceEvent)
+        
+        resetTestExpectations()
+        networkService.reset()
+        
+        for locationHint in (EdgeLocationHint.allCases).map({ $0.rawValue }).shuffled() {
+            Edge.setLocationHint(locationHint)
+            
+            expectEdgeEventHandle(expectedHandleType: TestConstants.EventSource.STATE_STORE, expectedCount: 1)
+            
+            // Test
+            Edge.sendEvent(experienceEvent: experienceEvent)
+            
+            // Verify
+            let expectedStateStoreJSON = #"""
+            {
+              "payload": [
+                {
+                  "maxAge": 123,
+                  "key": "kndctr_972C898555E9F7BC7F000101_AdobeOrg_cluster",
+                  "value": "\#(locationHint)"
+                }
+              ]
+            }
+            """#
+            
+            // See testSendEvent_receivesExpectedEventHandles for existence validation
+            let stateStoreEvent = getEdgeEventHandles(expectedHandleType: TestConstants.EventSource.STATE_STORE).last!
+            
+            // Exact match used here to strictly validate `payload` array element count == 2
+            assertExactMatch(expected: getAnyCodable(expectedStateStoreJSON)!,
+                            actual: getAnyCodable(stateStoreEvent),
+                            typeMatchPaths: ["payload[0].maxAge"])
+            
+            resetTestExpectations()
+            networkService.reset()
+        }
+    }
+    
+
+    // MARK: 2nd event tests
+    func testSendEventx2_receivesExpectedNetworkResponse() {
+        // Setup
+        // These expectations are used as a barrier for the event processing to complete
+        expectEdgeEventHandle(expectedHandleType: TestConstants.EventSource.LOCATION_HINT_RESULT)
+
+        let experienceEvent = ExperienceEvent(xdm: ["xdmtest": "data"],
+                                              data: ["data": ["test": "data"]])
+        
+        Edge.sendEvent(experienceEvent: experienceEvent)
+        
+        // Extract location hint from Edge Network location hint response event
+        guard let locationHintResult = getLastLocationHintResultValue() else {
+            XCTFail("Unable to extract valid location hint from location hint result event handle.")
+            return
+        }
+        
+        // Wait on all expectations to finish processing before clearing expectations
+        assertExpectedEvents(ignoreUnexpectedEvents: true)
+        
+        // Reset all test expectations
+        networkService.reset()
+        resetTestExpectations()
+        
+        // Set actual testing expectations
+        // If test suite level location hint is not set, uses the value extracted from location hint result
+        let locationHintNetworkRequest = NetworkRequest(urlString: createURLWith(locationHint: edgeLocationHint?.rawValue ?? locationHintResult), httpMethod: .post)!
```

Comment:
can be simplified as the for second event should always use locationHintResult (which may be equal to the initial edgeLocationHint if set)